src/main/java/mil/dds/anet/search/AbstractAttachmentSearcher.java

```diff
@@ -0,0 +1,57 @@
+package mil.dds.anet.search;
+
+import mil.dds.anet.beans.Attachment;
+import mil.dds.anet.beans.Person;
+import mil.dds.anet.beans.lists.AnetBeanList;
+import mil.dds.anet.beans.search.AttachmentSearchQuery;
+import mil.dds.anet.beans.search.ISearchQuery;
+import mil.dds.anet.database.AttachmentDao;
+import mil.dds.anet.database.mappers.AttachmentMapper;
+import mil.dds.anet.utils.DaoUtils;
+import ru.vyarus.guicey.jdbi3.tx.InTransaction;
+
+public abstract class AbstractAttachmentSearcher
+    extends AbstractSearcher<Attachment, AttachmentSearchQuery> implements IAttachmentSearcher {
+
+  public AbstractAttachmentSearcher(
+      AbstractSearchQueryBuilder<Attachment, AttachmentSearchQuery> qb) {
+    super(qb);
+  }
+
+  @InTransaction
+  @Override
+  public AnetBeanList<Attachment> runSearch(AttachmentSearchQuery query, Person user) {
+    buildQuery(query, user);
+    return qb.buildAndRun(getDbHandle(), query, new AttachmentMapper());
+  }
+
+  @Override
+  protected void buildQuery(AttachmentSearchQuery query) {
+    throw new UnsupportedOperationException();
+  }
+
+  protected void buildQuery(AttachmentSearchQuery query, Person user) {
```

Comment:
This method should be removed; do everything you need in the standard `buildQuery` method.